src/awkward/_util.py

```diff
@@ -37,6 +39,10 @@ def is_file_path(x):
         return False
 
 
+def is_sized_iterable(obj):
+    return isinstance(obj, Iterable) and isinstance(obj, Sized)
```

Comment:
Correct: too often, I assume that iterables have a `__len__` and common test examples don't check it. We're talking about `Sized` `Iterables` in anything we'd support in a slice tuple.

Reply:
It's a slight pain that there's not an ABC for sized iterables, but it's not too hard to write our own helper, so I should not complain :P